cmd/talosctl/cmd/mgmt/cluster/create.go

```diff
@@ -404,7 +404,9 @@ func create(ctx context.Context) (err error) {
 			fallthrough
 		case forceEndpoint != "":
 			endpointList = []string{forceEndpoint}
+			// using non-default endpoints, provision additional cert SANs and fix endpoint list
 			provisionOptions = append(provisionOptions, provision.WithEndpoint(forceEndpoint))
+			genOptions = append(genOptions, generate.WithAdditionalSubjectAltNames(endpointList))
```

Comment:
only for forced endpoint (usually docker cluster with `--endpoint=127.0.0.1` or other host IP), inject that IP as `certSAN` as Talos node doesn't see the endpoint (it's external IP essentially).